BuildResidentialHPXML/measure.rb

```diff
@@ -2963,169 +2963,113 @@ def validate_arguments(args)
-    errors << "cooling_system_type=#{args[:cooling_system_type]} and heat_pump_type=#{args[:heat_pump_type]}" if error
+    errors << 'Multiple central cooling systems are not currently supported.' if error
 
-    # non integer number of bathrooms
     if args[:geometry_unit_num_bathrooms] != Constants.Auto
       error = (Float(args[:geometry_unit_num_bathrooms]) % 1 != 0)
-      errors << "geometry_unit_num_bathrooms=#{args[:geometry_unit_num_bathrooms]}" if error
+      errors << 'Currently only supporting an integer number of bathrooms.' if error
     end
 
-    # non integer ceiling fan quantity
     if args[:ceiling_fan_quantity] != Constants.Auto
       error = (Float(args[:ceiling_fan_quantity]) % 1 != 0)
-      errors << "ceiling_fan_quantity=#{args[:ceiling_fan_quantity]}" if error
+      errors << 'Currently only supporting an integer quantity of ceiling fans.' if error
     end
 
-    # single-family, slab, foundation height > 0
-    warning = [HPXML::ResidentialTypeSFD, HPXML::ResidentialTypeSFA].include?(args[:geometry_unit_type]) && (args[:geometry_foundation_type] == HPXML::FoundationTypeSlab) && (args[:geometry_foundation_height] > 0)
-    warnings << "geometry_unit_type=#{args[:geometry_unit_type]} and geometry_foundation_type=#{args[:geometry_foundation_type]} and geometry_foundation_height=#{args[:geometry_foundation_height]}" if warning
+    warning = (args[:geometry_foundation_type] == HPXML::FoundationTypeSlab) && (args[:geometry_foundation_height] > 0)
+    warnings << 'Specified slab or above apartment foundation type with a non-zero foundation height. Assuming foundation height is zero.' if warning
 
-    # single-family, non slab, foundation height = 0
-    error = [HPXML::ResidentialTypeSFD, HPXML::ResidentialTypeSFA].include?(args[:geometry_unit_type]) && (args[:geometry_foundation_type] != HPXML::FoundationTypeSlab) && (args[:geometry_foundation_height] == 0)
-    errors << "geometry_unit_type=#{args[:geometry_unit_type]} and geometry_foundation_type=#{args[:geometry_foundation_type]} and geometry_foundation_height=#{args[:geometry_foundation_height]}" if error
+    error = ![HPXML::FoundationTypeSlab, HPXML::FoundationTypeAboveApartment].include?(args[:geometry_foundation_type]) && (args[:geometry_foundation_height] == 0)
+    errors << 'Specified a non-slab or non-above apartment foundation type with a height of zero.' if error
 
-    # single-family attached, multifamily and ambient foundation
     error = [HPXML::ResidentialTypeSFA, HPXML::ResidentialTypeApartment].include?(args[:geometry_unit_type]) && (args[:geometry_foundation_type] == HPXML::FoundationTypeAmbient)
-    errors << "geometry_unit_type=#{args[:geometry_unit_type]} and geometry_foundation_type=#{args[:geometry_foundation_type]}" if error
+    errors << 'Ambient foundation type for single-family attached or apartment units is not currently supported.' if error
```

Comment:
How much work would it be to allow this as we do for SFD?

Reply:
Created a separate issue: https://github.com/NREL/OpenStudio-HPXML/issues/945.